src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -324,6 +324,54 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <Target Name="AddDepsJsonAndRuntimeConfigToCopyItemsForReferencingProjects"
+          BeforeTargets="GetCopyToOutputDirectoryItems;_GetCopyToOutputDirectoryItemsFromThisProject"
+          Condition="'$(HasRuntimeOutput)' == 'true'">
+
+    <ItemGroup Condition="'$(GenerateDependencyFile)' == 'true'">
+      <AllItemsFullPathWithTargetPath Include="$(ProjectDepsFilePath)"
+                                      TargetPath="$([System.IO.Path]::GetFileName($(ProjectDepsFilePath)))"
+                                      CopyToOutputDirectory="PreserveNewest" />
+    </ItemGroup>
+    
+    <ItemGroup Condition=" '$(GenerateRuntimeConfigurationFiles)' == 'true'">
+      <AllItemsFullPathWithTargetPath Include="$(ProjectRuntimeConfigFilePath)"
+                                TargetPath="$([System.IO.Path]::GetFileName($(ProjectRuntimeConfigFilePath)))"
+                                CopyToOutputDirectory="PreserveNewest" />
+      <AllItemsFullPathWithTargetPath Include="$(ProjectRuntimeConfigDevFilePath)"
+                                TargetPath="$([System.IO.Path]::GetFileName($(ProjectRuntimeConfigDevFilePath)))"
+                                CopyToOutputDirectory="PreserveNewest" />
+    </ItemGroup>
+    
+  </Target>
+
+  <Target Name="AddDepsJsonAndRuntimeConfigToPublishItemsForReferencingProjects"
```

Comment:
No other way but need to write the same logic twice for publish and build?

And I don't see the logic of for copying the the referenced exe, how that worked?

Reply:
The logic is not quite the same.  The files can be different between build and publish in some cases.

For copying the referenced exe: The apphost is added as a `None` item with `CopyToOutputDirectory=PreserveNewest` in the `_ComputeNETCoreBuildOutputFiles`, with `Link` set to the target filename.  That gets gathered by the targets that get items to copy to the output directory from referenced projects.